File: backend/controllers/userController.js
const { v4: uuidv4 } = require('uuid');

const createUser = async (req, res) => {
  const user = { ...req.body, id: uuidv4() };
  // Save to DynamoDB
  res.status(201).json({ message: 'User registered', user });
};

module.exports = { createUser };

File: backend/controllers/user_controller.go
package controllers

import (
	"encoding/json"
	"net/http"

	"github.com/google/uuid"
)

func CreateUser(w http.ResponseWriter, r *http.Request) {
	var payload map[string]any
	if err := json.NewDecoder(r.Body).Decode(&payload); err != nil {
		writeJSON(w, http.StatusBadRequest, map[string]string{"error": "invalid request body"})
		return
	}

	payload["id"] = uuid.NewString()
	writeJSON(w, http.StatusCreated, map[string]any{
		"message": "User registered",
		"user":    payload,
	})
}
